### PR TITLE
JSON-LD Langmaps

### PR DESCRIPTION
**Pull request by lbiedinger**

### CODE REVIEW

File: tests/unit/components/entity/EntityDetails.spec.js [hunk 1]
```diff
@@ -32,32 +32,33 @@ describe('components/entity/EntityDetails', () => {
...
 
     wrapper.findAll('img').length.should.eq(1);
-    wrapper.text().should.not.contain(entityDetails.description);
+    wrapper.text().should.not.contain(entityDetails.description.values[0]);
   });
 
   it('does not show a show more link', () => {
     const wrapper = factory(entityDetails);
-
+    console.log(wrapper.html());
```
Comment (mirjamverloop) [inline tests/unit/components/entity/EntityDetails.spec.js]: Remove this console.log.

File: components/entity/EntityDetails.vue [hunk 2]
```diff
@@ -64,9 +65,10 @@
         type: String,
         default: ''
       },
+      // Description as object with 'values'(array of strings) and 'code' two letter language code
```
Comment (rwd) [inline components/entity/EntityDetails.vue]: Add a space before the parentheses.

File: components/entity/EntityDetails.vue [hunk 3]
```diff
@@ -25,20 +25,21 @@
         {{ title }}
       </h1>
       <p
-        v-if="description"
+        v-if="hasDescription"
         data-qa="entity description"
+        :lang="description.code"
       >
-        {{ showAll ? description : truncatedDescription }}
+        {{ showAll ? description.values[0] : truncatedDescription }}
```
Comment (rwd) [inline components/entity/EntityDetails.vue]: Add a new computed property `fullDescription` which returns `description.values[0]`

File: pages/entity/_type/_.vue [hunk 4]
```diff
@@ -125,7 +125,7 @@
         return this.editorialDepiction ? this.$t('goToRecord') : this.$t('entityDepictionCredit');
       },
       description() {
-        return this.editorialDescription ? this.editorialDescription : entities.getEntityDescription(this.entity);
+        return this.editorialDescription ? { values: [this.editorialDescription], code: null } : entities.getEntityDescription(this.entity, this.$i18n.locale);
```
Comment (rwd) [inline pages/entity/_type/_.vue]: This results in malformed meta tag content:
```html
<meta data-n-head="ssr" data-hid="og:description" property="og:description" content="[object Object]">
```

File: plugins/europeana/entity.js [hunk 5]
```diff
@@ -296,21 +296,22 @@ function getRelatedEntityTitleLink(entities) {
...
-    description = entity.note.en ? entity.note.en[0] : '';
+    description = langMapValueForLocale(entity.note, locale);
   } else if (entity.type === 'Agent' && entity.biographicalInformation) {
     // check if biographicalInformation is an array of objects
     // TODO: it _should_ always be an array. this is an Entity API bug. remove
     //       the condition when fixed upstream.
     //       see: https://europeana.atlassian.net/browse/EA-1685
     if (entity.biographicalInformation.length !== undefined) {
-      description = entity.biographicalInformation.filter(info => info['@language'] === 'en')[0]['@value'];
+      description = langMapValueForLocale(entity.biographicalInformation, locale);
```
Comment (rwd) [inline plugins/europeana/entity.js]: This does not appear to correctly handle the malformed case where biographicalInformation is not an array, e.g. /en/entity/person/37430-antonio-monticini

File: pages/entity/_type/_.vue [hunk 6]
```diff
@@ -125,7 +126,10 @@
...
+      },
+      descriptionText() {
+        return (this.description && this.description.values.length >= 1) ? this.description.values[0] : `${this.title} ${this.$t('entity')}`;
```
Comment (rwd) [inline pages/entity/_type/_.vue]: The fallback for description meta tags when no actual description is present is not a useful, i.e. descriptive, one. Let's omit those tags when there is no description.

**Pull request merged**